src/Kuzzle.ts

```diff
@@ -815,4 +846,26 @@ Discarded request: ${JSON.stringify(request)}`));
 
     dequeuingProcess();
   }
+
+  /**
+   * Sends a request with a timeout
+   */
+  private _timeoutRequest(delay, {request, options = {}}) {
+    // No timeout
+    if (delay === -1) {
+      return this.protocol.query(request, options);
+    }
+
+    const timeout = new Promise((resolve, reject) => {
+      const id = setTimeout(() => {
+        clearTimeout(id);
```

Comment:
I'm not sure you need to clear the timeout. Usually you do this to cancel a timeout before execution but once it's executed Node.js will clear the internal timers itself